phoenix-android/src/main/kotlin/fr/acinq/phoenix/android/security/EncryptedData.kt

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2025 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.phoenix.android.security
+
+import fr.acinq.bitcoin.ByteVector
+import fr.acinq.bitcoin.Crypto
+import fr.acinq.bitcoin.KeyPath
+import fr.acinq.bitcoin.byteVector
+import fr.acinq.lightning.crypto.ChaCha20Poly1305
+import fr.acinq.lightning.crypto.LocalKeyManager
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
+import java.util.zip.ZipOutputStream
+import javax.crypto.Cipher
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+
+/**
+ * This object represents data encrypted with a key derived from the wallet seed. The type of encryption and key used depends on [version].
+ *
+ * Note that the data is zipped before encryption.
+ *
+ * @param version the version used for the data encryption and for serializing this object
+ * @param data a byte array containing the zipped & encrypted payload, followed by the stuff needed for decryption (which depends on the version)
+ */
+class EncryptedData(val version: Version, val data: ByteArray) {
```

Comment:
It looks that the use case here is very similar to the encrypted channels backup extensions we added to some LN messages. We should probably export encryption methods from `lightning-kmp` instead of duplicating the code (and it would take care of private key handling) ?

Reply:
Yes, it is heavily inspired by that code, but I don't think it should be moved to lightning-kmp in a generic encryption helper:
- the key used is specific to Phoenix (and in fact we now use the `cloudKey` key already used by the iOS app for cloud backups), so it should stay in the Phoenix repo
- Most of the code in `EncryptedData` is related to serialisation and zipping the data. The code specific to encryption/decryption is actually very small. Really it's only how we compute the nonce, and then calling `encrypt/decrypt`. We would not save much in Phoenix, and it would pollute lightning-kmp.